lib/modules/versioning/hashicorp/convertor.ts

```diff
@@ -1,11 +1,18 @@
 import { logger } from '../../../logger';
 import { regEx } from '../../../util/regex';
 
+// Taken from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string. Licensed under CC BY 3.0
+// Removed the ^ and $.
+// Made minor and patch versions optional by surrounding them in parentheses followed by a question mark.
+const semverRegex =
```

Comment:
use `!!semver.valid(value)` instead

Reply:
I see the point and initially agreed, that that is the cleaner option.
However, while trying to get this work, I noticed a problem:

`semver.valid` does not match partial semver like (`1.0`) even with `loose: true`.
We would need to use `semver.valid(semver.corece(value))`.
However, `semver.coerce` is much more liberal then just allowing partial semver.
I am not sure if we really want this.

What do you think?
1. Should we use `semver.coerce` here?
2. If we use `semver.coerce`: Should we return the coerced semver or the raw input?
3. Do you have any other ideas on using the semver package without needing to coerce?

Reply:
still not happy 🤔